Log the list of tests in a given batch

jasondamour opened the pull request:
This PR logs the list of test files in a given batch before tests execute.

The batch listing is logged at `debug` level using a named logger (`pants.backend.python.goals.pytest_runner`), which integrates with the existing `--log-levels-by-target` mechanism. To surface it in CI, add to `pants.toml`:

```toml
[GLOBAL]
log_levels_by_target = {"pants.backend.python.goals.pytest_runner" = "info"}
```

Example output when enabled:
```
09:59:41.68 [INFO] batch of a_test.py:tests and 3 other files contains:
  a_test.py:tests
  b_test.py:tests
  c_test.py:tests
  d_test.py:tests
```

jasondamour committed:
Log the list of tests in a given batch

benjyw commented:
Nice! I suppose this should be debug by default but turned to info specifically in CI config. 

You can check for the relevant logging target by setting `--show-log-target`, which will then show the names in the log lines.

cognifloyd commented:
I think a configurable log level in `pants.toml` is the way to go. Your naming of `[test].batch_log_level` is fine.

I wonder if people will want to start tuning the level of other messages. You could make it extensible by adding something like `[test.log_levels_by_message]` (though that might be over engineering at this point):

```toml
[test.experimental_log_levels_by_message]
files_in_batch = 'info'
```

benjyw commented:
> I think a configurable log level in `pants.toml` is the way to go. Your naming of `[test].batch_log_level` is fine.

I am in friendly disagreement. Why single this feature out for special log configuration when we already have a robust global mechanism for adjusting the log level on a component-by-component basis?

jasondamour commented:
I did a quick test with `--show-log-target`, and unfortunately it seems like all of the "test process" logic is part of `workunit_store` log target, which isn't the level of granularity i was hoping for to single out this log event

benjyw commented:
> I did a quick test with `--show-log-target`, and unfortunately it seems like all of the "test process" logic is part of `workunit_store` log target, which isn't the level of granularity i was hoping for to single out this log event

Can we not introduce a new log target though? I haven't spelunked the code to see how to do that, but I have to hope it's straightforward...

benjyw commented:
Gently checking in on the status of this...?

sureshjoshi commented:
Hey @jasondamour  Just checking in on the status of this? Still has value?

jasondamour commented:
Rebased and updated. Addressed the feedback by using `logging.getLogger(__name__)` instead of the root logger — this gives the module a distinct log target (`pants.backend.python.goals.pytest_runner`) that plugs into the existing `--log-levels-by-target` mechanism.

The batch listing now logs at `debug` by default. To surface it in CI, add to `pants.toml`:

```toml
[GLOBAL]
log_levels_by_target = {"pants.backend.python.goals.pytest_runner" = "info"}
```

benjyw commented:
I've marked this as "release notes not required" - seems like overkill to mention a bit of debug-level logging in the release notes.

jasondamour commented:
Thanks, anything left to get merged?

benjyw commented:
Thanks @jasondamour !
